### PR TITLE
Clear debounced text field once form is submitted

### PR DESCRIPTION
**Pull request by hbulgarini**
DebouncedTextField component is not getting its local value reseted once the reducer state gets reseted. This is expected because the value for the input should be independent  from the reducer value since the reducer one might contain different value as the one shown in the input component ( i.e `transferInput`).

This PR resets the input properly once the form gets executed cleaning everything. 
Also this PR solves some no desired validation messages on some inputs while the transaction is submited.

### CODE REVIEW

File: src/hooks/react/useDebounceState.ts [hunk 1]
```diff
@@ -49,8 +60,28 @@ export const useDebounceState = ({ initialValue, wait = 500, transformCallback,
...
+    if (toReset && reset) {
+      setValue('');
+      setDebounced(null);
+      setToReset(false);
+    }
+
+    if (!shouldDispatch && !reset && previousReset !== reset) {
+      setShouldDispatch(true);
+    }
+  }, [reset, previousReset, setValue, value, debounced, toReset, shouldDispatch]);
```
Comment (wirednkod) [inline src/hooks/react/useDebounceState.ts:84]: I am not sure I understand the reasoning behind the existence of `toReset` and `previousReset`. I think the code can be much more simplified with the one below:
```suggestion
  useEffect(() => {
    if (reset) {
      if (value && debounced) {
        setShouldDispatch(false);
      }
      setValue('');
      setDebounced(null);
    }

    if (!shouldDispatch && !reset) {
      setShouldDispatch(true);
    }
  }, [reset, setValue, value, debounced, shouldDispatch]);
```

I am sure I am missing some reasoning here - but its a good conversation starter

**Review by wirednkod — CHANGES_REQUESTED** [review]

File: src/hooks/react/useDebounceState.ts (hunk 1 above)
Comment (tomusdrw) [reply src/hooks/react/useDebounceState.ts:84]: Yeah, I concur. The whole `setToReset` seems very unncessary. We should basically only check that `reset && !prevReset === true` and then run the clean up logic (i.e. the `reset` value changed from `false -> true`).

File: src/components/DebouncedTextField.tsx [hunk 2]
```diff
@@ -33,6 +33,7 @@ interface Props {
   rows?: number;
   dispatchCallback?: (value: ValueType) => void;
   initialValue?: ValueType;
+  reset?: boolean;
```
Comment (tomusdrw) [inline src/components/DebouncedTextField.tsx:36]: This smells of incorrect split of responsibilities. If we need to control internal state of the component from the outside it probably means that the state is not really that local.

I'd rather suggest to have `transactionId` being either passed as a prop or obtained from a reducer. Every time we send transaction `transactionId` is incremented, and all components are able to do any clean up they want if they notice that `transactionId` changes. On the first sight this looks much cleaner to me than the `reset/setToReset/toRest/previousReset` mumbojumbo happening internally now and is probably more future proof than passing `reset` to some specific components.

File: src/reducers/transactionReducer.ts [hunk 3]
```diff
@@ -90,6 +97,15 @@ export default function transactionReducer(state: TransactionState, action: Tran
...
       const { customCallInput, createType, targetChain } = action.payload;
+      if (!customCallInput) {
+        return {
+          ...state,
+          customCallInput,
+          transactionReadyToExecute: false,
+          customCallError: null,
+          shouldEvaluatePayloadEstimatedFee: false
+        };
+      }
```
Comment (tomusdrw) [inline src/reducers/transactionReducer.ts:108]: How is this related?

File: src/reducers/transactionReducer.ts [hunk 4]
```diff
@@ -64,6 +63,14 @@ export default function transactionReducer(state: TransactionState, action: Tran
...
     case TransactionActionTypes.SET_TRANSFER_AMOUNT: {
       const { transferAmount, chainDecimals } = action.payload;
+      if (!transferAmount) {
+        return {
+          ...state,
+          transferAmount,
+          transferAmountError: null,
+          transactionReadyToExecute: false
+        };
+      }
```
Comment (tomusdrw) [inline src/reducers/transactionReducer.ts:73]: How is this related?

File: src/hooks/react/useDebounceState.ts (hunk 1 above)
Comment (hbulgarini) [reply src/hooks/react/useDebounceState.ts]: previous reset is not needed as we discussed. I have improved the code based on your suggestion.

File: src/components/DebouncedTextField.tsx (hunk 2 above)
Comment (hbulgarini) [reply src/components/DebouncedTextField.tsx:36]: shit, this is  such a great suggestion .... asking the transactionId to be different is such a greater event than transaction changed from false to true.

File: src/hooks/react/useDebounceState.ts (hunk 1 above)
Comment (hbulgarini) [reply src/hooks/react/useDebounceState.ts]: guys i will resolve the conversation because i will base the process on listening for transaction.id changes, so this discussion gets outdated.

File: src/reducers/transactionReducer.ts (hunk 3 above)
Comment (hbulgarini) [reply src/reducers/transactionReducer.ts:108]: When inputs are cleared ( because of this process or manual ) there is an error message which in my opinion is unnecessary.  The expected behavior is that when input gets cleared for whatever reason the execution button should not be available, the fee should not be calculated and no error message should appear on the input.

File: src/reducers/transactionReducer.ts (hunk 4 above)
Comment (hbulgarini) [reply src/reducers/transactionReducer.ts:73]: https://github.com/paritytech/parity-bridges-ui/pull/248#discussion_r679939981

File: src/reducers/transactionReducer.ts (hunk 3 above)
Comment (hbulgarini) [reply src/reducers/transactionReducer.ts:108]: I think it is related in the way that is part of the reset process which is the whole point of this PR, even if it's not 100% related to debounce text field component.

**Review by wirednkod — APPROVED** [review]
